infrastructure/src/main/java/org/wso2/testgrid/infrastructure/providers/AWSProvider.java

```diff
@@ -205,8 +210,12 @@ private InfrastructureProvisionResult doProvision(InfrastructureConfig infrastru
 
             return result;
         } catch (WaiterUnrecoverableException e) {
-            throw new TestGridInfrastructureException(StringUtil.concatStrings("Error while waiting for stack : "
-                    , stackName, " to complete"), e);
+            String msg = getFailedStackEvents(stackName, cloudFormation);
```

Comment:
Please log the failure exception as the first item.

Reply:
What will happen if getFailedStackEvents threw an exception, say a RuntimeException!